cf-agent/verify_files_utils.c

```diff
@@ -829,10 +832,13 @@ static PromiseResult SourceSearchAndCopy(EvalContext *ctx, const char *from, cha
         strlcpy(newfrom, from, sizeof(newfrom) - 2); /* Assemble pathname */
         strlcpy(newto, to, sizeof(newto) - 2);
 
-        if (!JoinPath(newfrom, dirp->d_name))
+        if (!PathAppend(newfrom, sizeof(newfrom), dirp->d_name, sep))
         {
+            Log(LOG_LEVEL_ERR, "Internal limit reached in SourceSearchAndCopy(),"
+                " source path too long: '%s' + '%s'",
+                newfrom, dirp->d_name);
             AbstractDirClose(dirh);
-            return result;
+            return result;                             /* TODO return FAIL? */
```

Comment:
I'm wondering what's the meaning of *not* failing the promise at this point. This and a couple more TODO marks in this pull request should be changed ASAP IMHO, it's just that I didn't want to do it during the code freeze.